Pester.psm1

```diff
@@ -659,6 +659,40 @@ function Get-ScriptBlockScope
     [scriptblock].GetProperty('SessionStateInternal', $flags).GetValue($ScriptBlock, $null)
 }
 
+function Get-OperatingSystem
+{
+    [CmdletBinding()]
+    param()
+
+    if ($PSVersionTable.PSVersion.Major -lt 6 -or $IsWindows)
```

Comment:
A comment explaining why we do the check this way would be warranted here. If you don't know that PowerShell was windows only up until version 6 it might be a bit confusing.